### PR TITLE
Adding axis option to from_entry_expr methods

### PR DESCRIPTION
**Pull request by konradjk**

### CODE REVIEW

File: hail/python/hail/linalg/blockmatrix.py [hunk 1]
```diff
@@ -666,15 +669,27 @@ def write_from_entry_expr(entry_expr, path, overwrite=False, mean_impute=False,
         else:
             n_cols = mt.count_cols()
```
Comment (daniel-goldstein) [inline hail/python/hail/linalg/blockmatrix.py]: Shouldn't this depend on the axis?

Comment (konradjk) [reply hail/python/hail/linalg/blockmatrix.py]: Yep, good catch

File: hail/python/hail/linalg/blockmatrix.py [hunk 2]
```diff
@@ -664,17 +670,30 @@ def write_from_entry_expr(entry_expr, path, overwrite=False, mean_impute=False,
...
+            compute = {
+                '__count': agg.count_where(hl.is_defined(mt['__x'])),
+                '__sum': agg.sum(mt['__x']),
+                '__sum_sq': agg.sum(mt['__x'] * mt['__x'])
+            }
+            if axis == 'rows':
+                n_cols = mt.count_cols()
+                mt = mt.select_rows(**compute)
+            else:
+                n_cols = mt.count_rows()
```
Comment (daniel-goldstein) [inline hail/python/hail/linalg/blockmatrix.py]: Looks good, but I think the name `n_cols` is misleading here. Maybe something like `total_count`?

Comment (konradjk) [reply hail/python/hail/linalg/blockmatrix.py]: Yeah, I wondered about that. `n_elements` maybe?

Comment (daniel-goldstein) [reply hail/python/hail/linalg/blockmatrix.py]: I think that sounds good

**Comment by konradjk** [comment]
Also, I just tested this out with my own jar/zip and it does exactly what I wanted it to! So at least there's one empirical test

**Pull request merged**